src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -35,7 +35,11 @@ internal class MissingSetupComponentRegistrationService : OnceInitializedOnceDis
 
     private static readonly ImmutableHashSet<string> s_supportedReleaseChannelWorkloads = ImmutableHashSet.Create(StringComparers.WorkloadNames, WasmToolsWorkloadName);
 
+    // Lock objects
     private readonly object _webComponentIdsDetectedLock = new();
+    private readonly object _displayPromptLock = new();
+    private readonly object _lock = new();
```

Comment:
My preference would be to have the lock objects grouped with the internal data structures they protect. Though at that point it would probably be better to group each set of data and its lock in a different type. I'm reviewing this one commit at a time so maybe that's where you're going with this. :-)

Reply:
Agreed. My goal was to remove all potential for synchronous contention, so I grouped them to make them stand out. In the end all the lock objects were removed anyway.